### ISSUE TITLE
Segfault on malformed CSV with variable length rows and carriage return newlines

### ISSUE THREAD

**Issue by begelundmuller** [duckdb/duckdb#5077]
### What happens?

We're seeing DuckDB fail with a segmentation fault when ingesting a CSV that has both:

- variable length rows
- carriage return newlines (`\r`)

I apologize for submitting such a ridiculous bug report, but we've actually encountered such a CSV in the wild.

### To Reproduce

This succeeds:

```bash
$ echo -n "a,b\nc,d,e" > foo.csv
$ duckdb -c "select * from 'foo.csv'"
```

This segfaults:

```bash
$ echo -n "a,b\rc,d,e" > foo.csv
$ duckdb -c "select * from 'foo.csv'"
```

### OS:

macOS 12.5.1

### DuckDB Version:

v0.5.1

### DuckDB Client:

CLI

### Full Name:

Benjamin Egelund-Müller

### Affiliation:

Rill Data

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by hannes** [comment]
In debug mode this trips an assertion 

```
Error: INTERNAL Error: Assertion triggered in file "/Users/hannes/source/duckdb/src/common/types/data_chunk.cpp" on line 77: index < size()
```

Stack trace

```
    frame #1: 0x000000010553fddc duckdb`duckdb::DuckDBAssertInternal(condition=false, condition_name="index < size()", file="/Users/hannes/source/duckdb/src/common/types/data_chunk.cpp", linenr=77) at assert.cpp:10:2
    frame #2: 0x0000000104be2d3c duckdb`duckdb::DataChunk::GetValue(this=0x000000016fdf0000, col_idx=0, index=0) const at data_chunk.cpp:77:2
    frame #3: 0x0000000106544160 duckdb`duckdb::BufferedCSVReader::DetectHeader(this=0x0000000119017480, best_sql_types_candidates=size=0, best_header_row=0x000000016fdf0000) at buffered_csv_reader.cpp:1175:38
    frame #4: 0x0000000106527dd8 duckdb`duckdb::BufferedCSVReader::SniffCSV(this=0x0000000119017480, requested_types=size=0) at buffered_csv_reader.cpp:1357:2
    frame #5: 0x0000000106524c98 duckdb`duckdb::BufferedCSVReader::Initialize(this=0x0000000119017480, requested_types=size=0) at buffered_csv_reader.cpp:527:15
```

**Comment by Mytherin** [comment]
Thanks for the report! I have pushed a fix in #5083
